pages/docs/guides/message-validation.md

```diff
@@ -0,0 +1,147 @@
+---
+title: Validate messages
+description: This guide explains different use cases for message (event) validation using AsyncAPI.
+weight: 120
+---
+
+# Introduction
+This guide will teach you how to validate the messages in runtime after creating your application using AsyncAPI.
+
+# Message validation
+To understand message validation, we must first understand the basic components involved.
+- Producer: The producer is responsible for producing messages.
+- Consumer: The consumer is responsible for getting the producer's messages.
+- Broker: The broker acts as a bridge between the consumer and the producer because messages travel through the broker.
+
+Both producers and consumers can do validation. Consumer-side validation happens by discarding invalid messages based on provided parameters.
+Because consumers and producers cannot communicate directly, the AsyncAPI file dictates what should be included in the payload when a service produces a message. The AsyncAPI file also tells the consumer about the message's properties.
+
+```mermaid
+stateDiagram-v2
+    Producer --> Broker : messages
+    Producer --> Fail : error
+    Broker --> Consumer : valid message
+    Broker --> Fail : invalid message
+```
+To validate AsyncAPI messages (events), we have two options: 
```

Comment:
there are more than 2 options and in this tutorial you describe 3 actually

I suggest you remove this sentence and links an also section `AsyncAPI schema validation`.
In previous comment I explained how you can introduce different validation methods

`AsyncAPI schema validation` and code samples from subsections are not needed. In all 3 scenarios it is all about validation of messages against schema, doesn't matter runtime or gateway.

you need 3 sections in this document:
- Validation in runtime or Validation in Consumer and/or Producer
   here you explain it with the diagram ☝🏼 and mention https://github.com/WaleedAshraf/asyncapi-validator as an example implementation, just short paragraph. you can also link to tutorial about validating messages where it is showcased. Explain that AsyncAPI document plays important role here as payload schemas are taken from AsyncAPI document as a source of truth
- Validation in a gateway (not AsyncAPI, AsyncAPI gateway is just an example, there might be others)
  here gateway diagram, and explain it is based on AsyncAPI gateway but yeah, others might be out there too. Explain that AsyncAPI document plays important role here as payload schemas are taken from AsyncAPI document as a source of truth
- Validation handled by the broker
   here "flow" diagram as well, explain that it depends on brokers implementations, that example you show is based on Kafka and Schema Registry concept. Write that AsyncAPI is not involved in validation process. But also write that payload schemas that are stored in Schema Registry can be referenced from AsyncAPI documents -> show example 

```yml
asyncapi: 2.0.0
info:
  title: Example with Avro
  version: 0.1.0
channels:
  example:
    publish:
      message:
        schemaFormat: 'application/vnd.apache.avro;version=1.9.0'
        payload:
          $ref: 'https://example.europe-west3.gcp.confluent.cloud/subjects/test/versions/1/schema'
```

Reply:
thank you for the review, I'll implement them